rfc/service-retrievals.md

```diff
@@ -0,0 +1,38 @@
+# Service Retrievals
+
+There are a number of occasions where internal services within the Storacha Network need to retrieve data from Storage Nodes:
+
+1. The _Upload Service_ retrieves indexes as part of an `space/index/add` invocation, so that it can publish all the hashes to IPNI.
+2. The _Indexing Service_ fetches blob indexes when an `assert/index` claim is published to it, as well as part of the normal flow when querying for hashes.
+3. The _Filecoin Service_ fetches all data submitted to it in order to validate the provided Piece CID matches the data.
```

Comment:
It's on the roadmap but for warm storage, the storage node itself has to make a piece CID, and we have the one from the client. I believe agreement of these piece CIDs is sufficient to throw it in a filecoin deal.

When I say "on the roadmap" -- it's somewhere in the October development I believe.

This doesn't cover one other use case, which is retrieval by the actual Filecoin SP